virtualsmartcard/win32/BixVReader/TcpIpReader.cpp

```diff
@@ -291,16 +312,18 @@ void TcpIpReader::shutdown() {
 	serverThread=NULL;
 	eventsocket=NULL;
 	socket=NULL;
-	if (waitRemoveIpr!=NULL) {
+	while (!waitRemoveIpr.empty()) {
 		SectionLocker lock(device->m_RequestLock);
-		if (waitRemoveIpr->UnmarkCancelable()==S_OK)
-			waitRemoveIpr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
-		waitRemoveIpr=NULL;
+		CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();
+		if (ipr->UnmarkCancelable()==S_OK)
+			ipr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
+		waitRemoveIpr.pop_back();
 	}
-	if (waitInsertIpr!=NULL) {
+	while (!waitInsertIpr.empty()) {
 		SectionLocker lock(device->m_RequestLock);
-		if (waitInsertIpr->UnmarkCancelable()==S_OK)
-			waitInsertIpr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
-		waitInsertIpr=NULL;
+		CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();

```

Comment:
Copy paste error. Should be `CComPtr<IWDFIoRequest> ipr = waitInsertIpr.back();`